pkg/yamlprocessor/yamlprocessor_test.go

```diff
@@ -221,9 +247,16 @@ foo:
 	}
 	for _, tc := range testcases {
 		t.Run(tc.name, func(t *testing.T) {
-			got, err := ReplaceValue([]byte(tc.yml), tc.path, tc.value)
-			assert.Equal(t, tc.wantErr, err != nil)
-			assert.Equal(t, tc.want, got)
+			p, err := NewProcessor([]byte(tc.yml))
+			require.NotNil(t, p)
+			require.NoError(t, err)
+
+			rerr := p.ReplaceString(tc.path, tc.value)
+			got, err := p.Bytes()
+			assert.Equal(t, tc.wantErr, err != nil || rerr != nil)
+			if !tc.wantErr {
+				assert.Equal(t, tc.want, got)
+			}
```

Comment:
```suggestion
			require.Equal(t, tc.wantErr, err != nil || rerr != nil)
			assert.Equal(t, tc.want, got)
```
nit

Reply:
The check is needed because `Bytes` function returns the original data when the operation fails.